core/src/domain/user/ports.rs

```diff
@@ -149,6 +149,10 @@ pub trait UserRoleService: Send + Sync {
         user_id: Uuid,
         role_id: Uuid,
     ) -> impl Future<Output = Result<bool, UserError>> + Send;
+    fn delete_role_relations_by_id(
+        &self,
+        role_id: Uuid,
+    ) -> impl Future<Output = Result<u64, UserError>> + Send;
```

Comment:
Not sure this is the right place or the right name for this fn, but it does the thing.